### PR TITLE
Docs: provide more context to no-eq-null

### PR DESCRIPTION
**Pull request by gfyoung**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

* Encourage use of the more powerful `eqeqeq` rule for linting equality checks.

* Indicate this rule is compatible with an analogous rule in `JSHint`.

xref https://github.com/eslint/eslint/pull/179
closes https://github.com/eslint/eslint/issues/14749

#### Is there anything you'd like reviewers to focus on?

N/A

cc @btmills

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

**Review by aladdin-add — CHANGES_REQUESTED** [review]
there is a linting issue in the md file. can you fix it please?

File: docs/rules/no-eq-null.md [hunk 1]
```diff
@@ -39,3 +39,11 @@ while (qux !== null) {
...
 }
 ```
+
+## Compatibility
+
+* **JSHint**: This rule corresponds to `eqnull` rule of JSHint.
+
+## When Not To Use It
+
+If you want to enforce type-checking operations in general, use the more powerful [eqeqeq](./eqeqeq) instead.
```
Comment (btmills) [inline docs/rules/no-eq-null.md:49]: This should fix that missing new line failure.

```suggestion
If you want to enforce type-checking operations in general, use the more powerful [eqeqeq](./eqeqeq) instead.

```

**Review by btmills — COMMENTED** [review]
LGTM as soon as CI passes. Thanks for this improvement @gfyoung!

**Comment by gfyoung** [comment]
@aladdin-add @btmills: Can you guys take another look?

**Review by snitin315 — APPROVED** [review]
LGTM, thanks for contributing.

**Review by aladdin-add — APPROVED** [review]

**Review by btmills — APPROVED** [review]

**Pull request merged**